### PR TITLE
Added basic xenstore support

### PR DESCRIPTION
**Pull request by Columpio**
1. Breakpoint on 49 line of src/arch/xen/kernel_init.c
2. Continue
3. print list
4. got: `$1 = "EACCES\000\000\362\313\000\000\363\313\000\000\364\313\000\000\365\313\000\000\000\000\000\000 \00 0\000"`

### CODE REVIEW

File: src/arch/xen/xenstore.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
+#include "xenstore.h"
```
Comment (AntonKozlov) [inline src/arch/xen/xenstore.c]: There is no copyright here. Have the source any? If so, please add it (and yours too)

File: src/arch/xen/kernel_init.c [hunk 2]
```diff
@@ -38,6 +42,12 @@ void xen_kernel_start(start_info_t * start_info) {
 	
 	init_events();
 
+
+	xenstore_init(start_info);
+	char list[32];
+	uint32_t len = xenstore_ls("/", list, 32);
+	printk("len = %d\n", len);
```
Comment (AntonKozlov) [inline src/arch/xen/kernel_init.c]: you're calling printk before `diag` initialized, which performed later in `kernel_start`

**Comment by Columpio** [comment]
@AntonKozlov I asked you not to merge this because this is testing branch. What I mean is that calling xenstore_ stuff in kernel_init.c file is totally wrong and is made for testing only.

File: src/arch/xen/kernel_init.c (hunk 2 above)
Comment (Columpio) [reply src/arch/xen/kernel_init.c]: thx

**Comment by AntonKozlov** [comment]
That's true, but it is easy to fix. Do you have objections for me merging https://github.com/AntonKozlov/embox/tree/pull-1149? It's this PR with my commit on top

**Comment by Columpio** [comment]
Seems to work ^_^

**Comment by AntonKozlov** [comment]
Was unable to compile, starting from unit init function incompatible type up to missing `test_assert*` declarations (definitions, actually). I addressed that in next commit by moving test function to separate test module. 

Please let me know if my modifications caused any harm. Thanks!

**Pull request merged**